tests/v2/validation/provisioning/clustertemplates/provisioning_cluster_template_test.go

```diff
@@ -0,0 +1,335 @@
+			Required: isRequiredQuestion,
+			Default:  "",
+			Type:     "string",
+		},
+	}
+)
+
+type ClusterTemplateProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	standardUserClient *rancher.Client
+	provisioningConfig *provisioninginput.Config
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	c.provisioningConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, c.provisioningConfig)
+
+	client, err := rancher.NewClient("", testSession) //c.client
+	require.NoError(c.T(), err)
+
+	c.client = client
+
+	if c.provisioningConfig.RKE1KubernetesVersions == nil {
+		rke1Versions, err := kubernetesversions.ListRKE1AllVersions(c.client)
+		require.NoError(c.T(), err)
+
+		c.provisioningConfig.RKE1KubernetesVersions = []string{rke1Versions[len(rke1Versions)-1]}
+	}
+
+	if c.provisioningConfig.CNIs == nil {
+		c.provisioningConfig.CNIs = []string{cniCalico}
+	}
+
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) createRke1ClusterTemplateAndTemplateRevisions(clusterTemplateRevisionConfig clustertemplates.ClusterTemplateRevisionConfig, answers map[string]string, members []mgmt.Member) *clustertemplates.ClusterTemplateConfig {
+	clusterTemplate, err := rke1.CreateRkeTemplate(c.client, members)
+	require.NoError(c.T(), err)
+
+	clusterTemplateRevision, err := rke1.CreateRkeTemplateRevision(c.client, clusterTemplateRevisionConfig, clusterTemplate.ID)
+	require.NoError(c.T(), err)
+
+	clusterTemplateClusterConfig := new(clustertemplates.ClusterTemplateConfig)
+	clusterTemplateClusterConfig.RKE1TemplateID = clusterTemplate.ID
+	clusterTemplateClusterConfig.RKE1TemplateRevisionID = clusterTemplateRevision.ID
+
+	if answers != nil {
+		clusterTemplateClusterConfig.RKETemplateAnswers = &mgmt.Answer{
+			Values: answers,
+		}
+	}
+	return clusterTemplateClusterConfig
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) testCreateRke1ClusterWithTemplateAndVerify(client *rancher.Client, clusterTempClusterConfig *rke1.ClusterTemplateConfig) *mgmt.Cluster {
+
+	rke1Provider := provisioning.CreateRKE1Provider(c.provisioningConfig.NodeProviders[0])
+	nodeTemplate, err := rke1Provider.NodeTemplateFunc(client)
+	require.NoError(c.T(), err)
+
+	clusterConfig := clusters.ConvertConfigToClusterConfig(c.provisioningConfig)
+
+	nodeAndRoles := []provisioninginput.NodePools{
+		provisioninginput.AllRolesNodePool,
+	}
+	clusterConfig.NodePools = nodeAndRoles
+
+	clusterObject, err := provisioning.CreateProvisioningRKE1ClusterWithClusterTemplate(client, rke1Provider, clusterTempClusterConfig, clusterConfig, nodeTemplate)
+	require.NoError(c.T(), err)
+
+	clusterConfig.KubernetesVersion = c.provisioningConfig.RKE1KubernetesVersions[0]
```

Comment:
this should be set before creating the cluster, not after

Reply:
no for cluster provisioning using a template we do not set the kubernetes version in the cluster config, it gets the value from the rke template. here I'm setting the kubernetes version as I'm making use of the VerifyRKE1 cluster and it accepts cluster config. this is to make sure the cluster created with the cluster template has the expected kubernetes version.